webapp/src/ts/services/select2-search.service.ts

```diff
@@ -58,8 +61,7 @@ export class Select2SearchService {
   }
 
   private query(params, successCb, failureCb, options, types) {
-    const currentQuery = params.data.q;
-
+    this.currentQuery = params.data.q;
```

Comment:
Hmm, wait, I see that the `this.currentQuery !== params.data.q` check is happening _async_ in the code-flow below (after the `searchService.search` call).  So, referencing back to your earlier comment, having this as a class field allows a newer search to preempt an earlier one.  But, since we now know it is possible to have multiple references to this service from multiple elements at the same time, I am concerned that having this as a class field is just inviting weird conflicts/race-conditions (perhaps around enketo auto-triggering updates to contact selectors as things get recalculated?).  

Maybe it would be better to just leave this as a local variable and then remove the unnecessary check below...

Reply:
Right I think you're correct here... but at the same time this local variable becomes irrelevant as it's just an alias for `params.data.q` and the checks below are always true
Maybe I'm missing something about the scope of this variable but the way I read it, it can never be unequal to `params.data.q`. I'll remove `currentQuery` and its associated checks and we'll see if the CI breaks :man_shrugging: but I expect all tests to keep passing